CONTRIBUTING.md

```diff
@@ -79,47 +80,93 @@ from Authentication import constants
+This folder contains source code of the CodeChecker `analyzer` and
+`build-logger`.
+
+The `build-logger` can be found under the `tools` folder. This can be used to
+capture the build process and generate a JSON compilation database.
+
+#### `bin`
+This folder contains entry points of the CodeChecker package such as
+`codechecker-version` and wrapper scripts.
+
+#### `codechecker_common`
+`codechecker_common` is a python package where all common CodeChecker related
+source code is found which are used by the analyzer and web part of
+CodeChecker.
+
+#### `config`
+This folder contains common configuration files such as
+`checker_severity_map.json`, `logger.conf` etc. which are used by the
+`analyzer` and the `web` part of CodeChecker.
+
+#### `docker`
+This folder contains docker related files.
+
+#### `docs`
+This folder contains documentation files for the CodeChecker.
+
+#### `scripts`
+This folder contains multiple scripts which are used at the build process of
+CodeChecker, gerrit integration, debug etc.
+
+#### `tools`
+This folder contains tools which are used by the `analyzer` and `web` part
+of the CodeChecker such as `plist-to-html` and `tu_collector`.
+
+#### `web`
+This folder contains source code of the CodeChecker web server and web client.
+
+### Entry points
+CodeChecker is organised into different entry-points based on different
 features of the program, such as logging, analysing, result storage, web
-server, etc. All these features have their own module under `libcodechecker`.
+server, etc.
 
-A CodeChecker feature having its entry point consists of a bare minimum of
+A CodeChecker feature having it's entry point consists of a bare minimum of
 two (2) things:
 
  * An entry point under `bin/codechecker-myfeature`.
  * The entry point's definition containing the feature's command-line help and
-   argument parser under `libcodechecker/myfeature.py`.
+   argument parser under `cmd/myfeature.py`.
+
+### Libraries
+CodeChecker contains multiple python packages:
+ * `codechecker_common`: used by the analyzer and web part of the CodeChecker.
+ * `codechecker_analyzer`: contains source code of the CodeChecker analyzer.
+ * `codechecker_web`: used by the WEB server and client.
+ * `codechecker_client`: contains source code of the CodeChecker WEB client.
+ * `codechecker_server`: contains source code of the CodeChecker WEB server.
 
 Additionally, you may use different Python files to store code for your library
-for better code organisation. Library code related to `myfeature` go into the
-`libcodechecker/myfeature` folder.
+for better code organisation. For example a server library code related to
+`myfeature` go into the `codechecker_server/myfeature` folder.
 
-If the library code is used between multiple subcommands, the file is put into
-`libcodechecker` itself. Please try to localise your library code in its own
-folder as much as possible.
+If the library code is used between analyzer and web, the file is put into
```

Comment:
I would not recommend to put things to common if not really necessary.
We should decrease the size of it.